brigade-worker/src/app.ts

```diff
@@ -130,6 +130,7 @@ export class App {
     // Run if an uncaught rejection happens.
     process.on("unhandledRejection", (reason: any, p: Promise<any>) => {
       this.logger.error(reason);
+      console.log("Unhandled rejection: " + reason);
```

Comment:
Just curious: do we know where the pre-existing `this.logger.error(reason);` is actually logged?

Reply:
That's why I left the comment to investigate the log levels a bit more.

Reply:
Ok, I have an update here.  While testing some work on the Brigade GitHub App, I noticed that in events created by the app (Provider: github), the log level was such that all of the `this.logger*` calls in this file _were_ being printed to the logs, whereas the are not by default when running events via `brig` (Provider: brigade-cli).

When I describe the corresponding pods for each, `BRIGADE_LOG_LEVEL` is empty for the former while it is `log` for the latter.

Therefore, using this hint for running our same test with `brig`, I can get the error to show properly if I set log level to an empty string (as it is when produced by Provider: github currently:

```
 $ brig run vdice/empty-testbed -l "" -f throw-error.js
Event created. Waiting for worker pod named "brigade-worker-01db3k51tbq748hrzckby1hhdg".
Build: 01db3k51tbq748hrzckby1hhdg, Worker: brigade-worker-01db3k51tbq748hrzckby1hhdg
prestart: no dependencies file found
[brigade] brigade-worker version: 1.0.0
[brigade:k8s] Creating PVC named brigade-worker-01db3k51tbq748hrzckby1hhdg
[brigade:app]   Error: This should be logged when failing the event handler.

  - script:4 EventRegistry.<anonymous>
    /etc/brigade/..2019_05_17_19_08_59.317237096/script:4:9

  - events.js:107 EventRegistry.fire
    [src]/[@brigadecore]/brigadier/out/events.js:107:14

  - brigadier.js:34 Object.fire
    /home/src/dist/brigadier.js:34:20

  - app.js:157 loadProject.then.then
    /home/src/dist/app.js:157:23


  - next_tick.js:189 process._tickCallback
    internal/process/next_tick.js:189:7


[brigade:app] error handler is cleaning up
[brigade:k8s] Destroying PVC named brigade-worker-01db3k51tbq748hrzckby1hhdg
error Command failed with exit code 1.
```

To be noted, if `brig`'s log level is set to _anything_ besides `""`, we don't see the logs.  For instance, you'd think `error` would show them:
```
 $ b run vdice/empty-testbed -l error -f throw-error.js
Event created. Waiting for worker pod named "brigade-worker-01db3kf62vv6grqpgse7sv4y99".
Build: 01db3kf62vv6grqpgse7sv4y99, Worker: brigade-worker-01db3kf62vv6grqpgse7sv4y99
prestart: no dependencies file found
[brigade] brigade-worker version: 1.0.0
error Command failed with exit code 1.
```

So perhaps the real issue is in `brig`?  Somehow the log levels not taking affect?  And/or why an empty string might default to actually showing more logs?

Reply:
I would absoutely be in favour for a real fix rather than an additional `console.log` - so feel free not to merge this if you have the fix, I'll probably _not_ get around to this for at least another week..

Reply:
A lot of time spent on investigation and understanding the flow of things, but I believe https://github.com/brigadecore/brigade/pull/915 is the fix we're looking for.  Let me know what you think, @radu-matei